tests/integration/test_charm.py

```diff
@@ -23,24 +25,45 @@
 async def test_build_and_deploy(
     ops_test: OpsTest,
     charm_base: str,
-    filesystem_client_charm,
-    nfs_server_proxy_charm,
-    cephfs_server_proxy_charm,
+    filesystem_client_charm: Awaitable[str | pathlib.Path],
+    nfs_server_proxy_charm: Awaitable[str | pathlib.Path],
+    cephfs_server_proxy_charm: Awaitable[str | pathlib.Path],
 ) -> None:
     """Build the charm-under-test and deploy it together with related charms.
 
     Assert on the unit status before any relations/configurations take place.
     """
-    nfs_info = bootstrap_nfs_server()
-    cephfs_info = bootstrap_microceph()
-    logger.info(f"Deploying {', '.join(CHARMS)}")
 
-    # Pack charms.
-    filesystem_client, nfs_server_proxy, cephfs_server_proxy = await asyncio.gather(
-        filesystem_client_charm, nfs_server_proxy_charm, cephfs_server_proxy_charm
-    )
+    async def deploy_cephfs_proxy():
+        """Deploy a MicroCeph cluster and pass its info to a new `cephfs_server_proxy_charm`."""
+        cephfs_server_proxy = await cephfs_server_proxy_charm
+        (cephfs_info, _) = await asyncio.gather(
```

Comment:
Do these values need to be encapsulated within parenthesis? Python allows you to unpack tuples without them:

 ```suggestion
        cephfs_info, _ = await asyncio.gather(
```

However, I do know that Python can get twisty curvey with futures and won't always return the expected result of the asynchronous operation.

Reply:
Nope, it's just a stylistic choice. I've been writing too much Rust recently 😅